### DISCUSSION TITLE
제페토 카메라 충돌 관련 문제

### DISCUSSION THREAD

**Discussion by IndiaInk10** [naverz/zepeto-studio-kor#924] [5. Q&A]
Map과 Trigger 레이어를 만들고 20번째부터 레이어 설정하고 충돌되야 하는 오브젝트들에게 Map 레이어를 지정했음에도 가끔씩 카메라 충돌이 안되는 문제가 있습니다.

제 개인적인 생각으로는 벽의 충돌체가 너무 얇게 되어 있어 그런게 아닐까 하는 생각이 드는데요…

문제점을 정확하게 파악을 못하겠네요 😭
도움이 필요합니다!

**Comment by YSHFM** [comment]
충돌체 크기를 이미지로 첨부해주시면 도움이 더 될 것 같습니다. 너무 얇으시면 안될 수도 있어요

**Reply by IndiaInk10** [reply]
<img width="545" alt="pipe_1" src="https://user-images.githubusercontent.com/49700162/175884858-7cbb22d1-a2e1-4f12-8a79-aea60ca05b3d.png">
<img width="551" alt="pipe_2" src="https://user-images.githubusercontent.com/49700162/175884866-577e9642-b525-4a23-b4cd-ace85ca0f633.png">

@YSHFM 
문제가 발생하는 부분이 다음과 같이 되어있습니다!

**Reply by YSHFM** [reply]
오....빈 원통형이신거죠? 굵기가 너무 얇아서 그런 것 같습니다
https://docs.unity3d.com/kr/2021.3/Manual/ExecutionOrder.html
제가 알기론 피직스 이벤트 갱신이랑 Gamelohic Update갱신이 다르기 때문에 fixedupdate가 아닌 상황에서 체크가 필요하다면, collision은 부적합 할 수 있습니다.

**Reply by IndiaInk10** [reply]
@YSHFM #191 에서 치자피즈님이 알려주신 Zepeto에서 제공하는 기능을 사용했는데 해당 내용도 위 링크의 내용을 통해 해결이 가능할까요?

**Reply by YSHFM** [reply]
제페토라도 유니티 내부 이벤트 함수 순서를 크게 벗어나지 않을 것 같습니다. #657 이전에 이런 이슈도 있었던 만큼 이벤트 순서가 바뀌지는 않을 것 같습니다.

따라서, Collsion크기를 늘리셔야하거나, 기기의 프레임을 벗어난 충돌은 체크하지 못할 수도 있기 때문에, 별도의 스크립트를 작성하셔야 되는 것으로 보입니다.
